actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -95,6 +87,14 @@ def location_if_redirected
           location = normalize_argument_to_redirection(@response.location)
           " redirect to <#{location}>"
         end
+
+        def code_with_name(code_or_name)
+          if RESPONSE_PREDICATES.values.include?("#{code_or_name}?".to_sym)
+            code_or_name = RESPONSE_PREDICATES.invert["#{code_or_name}?".to_sym]
+          end
+
+          AssertionResponse.new(code_or_name).code_and_name
```

Comment:
This line means that we're creating a new AssertionResponse for each time `code_with_name` is called. This happens twice in `generate_response_message`, once for the `expected` and once for the `actual`. It's not ideal to create another response object for the Actual, since we already have one, but if we didn't we'd have to add the code for converting between the response **code** and the response **name** into the actual `Response` object.